demos/exit/README.md

```diff
@@ -0,0 +1,38 @@
+# Exit
+
+In this demo we are going to show that threads can exit normally like C functions. So we can use functions like `thread_join` to let a thread to wait for some threads to exit. 
+
+To explain the demo, we need to introduce:
+* `make_args` (in `include/common/thread.h`)
+* `add_named_thread_wtih_args` (in `src/user/thread.c`)
+* `thread_join` (in `src/user/thread.c`)
+
+## `make_args`
+
+In AMT, we use `ThreadArgs` to pack arguments so we can pass these variables into the thread we want to create. 
+
+Since `ThreadArgs` comes with 4 variables, we need to create the variable before we pass it into the thread.
+
+`make_args` helps us to transform the assignment into a form like a function call and make the assignment of the `ThreadArgs` becomes more readable.
+
+## `add_named_thread_with_args`
+
+`add_named_thread_with_args` creates a thread with the given `name`, `args` and `worker` routine.
+
+The behaviour is just like what `add_thread_from_worker()` does, but the arguments of the threads created by `add_thread_from_worker()` are initialized to 0 while we can pass the arguments into the threads created by `add_named_thread_with_args`.
+
+## `thread_join`
+
+The `thread_join()` function waits for the thread specified by `tid` to terminate. When a thread calls this function, it keeps yielding until the target thread exits. 
+
+If the value of `state` is not NULL, then `thread_join()` copies the exit state of the target thread into the location pointed to by `state`.
+
+## Walkthrough
+
+For starters, we create 2 threads and pass `2` and `4` into the corresponding threads. After the creation, the `setup` thread jumps into the `counter()` function.
+
+In the `counter()` function, it uses `thread_join` to wait thread 1. Since the thread does not exit, the `setup` thread will repeatedly yield. 
+
+Thread 1 will reach its end after it yields twice. So the `counter` thread should receive the state of thread 1 as `finished`, and start to wait for thread 2. At this time, thread 2 has two more times to yield.
```

Comment:
You can remove the comma:
"So the `counter` thread should receive the state of thread 1 as `finished` and start to wait for thread 2."

"At this time, thread 2 has two yields left before it will exit."